solutions/joi-15-inheritance.mdx

```diff
@@ -0,0 +1,143 @@
+---
+id: joi-15-inheritance
+source: JOI 2015
+title: Inheritance
+author: Kevin Sheng
+---
+
+# Explanation
+
+At first glance, it seems Kruskal's would be more appropriate for this problem, so let's use that.
+
+Let's first give each child their own DSU data structure to store which cities are connected with their railroads,
+and then iterate through the railroads through decreasing order of price,
+giving each railroad to the first child that can accept it. This is effectively the same as going through
+each child one at a time and constructing a maximum spanning tree for each of them individually.
+
+However, linearly iterating through the entire list of DSUs would result in a
+runtime of about $\mathcal{O}(KM)$, which is too much given the bounds.
+
+To find the first child that can accept a railroad faster, let's try using binary search!
+But for binary search to actually work, we first need to prove the following:
+
+No matter what railroad we're processing,
```

Comment:
```suggestion
**Lemma:** No matter what railroad we're processing,
```

Reply:
hold on
aren't lemma's supposed to be part of a bigger proof?
what's the "bigger proof" then